source/background/library/contextMenu.js

```diff
@@ -0,0 +1,68 @@
+import { createNewTab, getCurrentTab, getExtensionURL, sendTabMessage } from "../../shared/library/extension.js";
+
+const CONTEXT_SHARED_ALL = {
+    contexts: ["all"]
+};
+const CONTEXT_SHARED_EDITABLE = {
+    contexts: ["editable"]
+};
+
+let __menu;
+
+export function updateContextMenu() {
+    chrome.contextMenus.removeAll();
+    __menu = chrome.contextMenus.create({
+        title: "Buttercup",
+        ...CONTEXT_SHARED_ALL
+    });
+    const unlockArchiveMenu = chrome.contextMenus.create({
+        title: "Unlock archive",
+        parentId: __menu,
+        ...CONTEXT_SHARED_ALL
+    });
+    chrome.contextMenus.create({
+        title: "Lock all archives",
+        parentId: __menu,
+        onclick: () => {
+            createNewTab(getExtensionURL("setup.html#/lock-archives"));
+        },
+        ...CONTEXT_SHARED_ALL
+    });
+    chrome.contextMenus.create({
+        parentId: __menu,
+        type: "separator",
+        ...CONTEXT_SHARED_ALL
+    });
+    chrome.contextMenus.create({
+        title: "Log in using",
+        parentId: __menu,
+        onclick: () => {
+            // todo
```

Comment:
Are we going to release with these in to-do state?

Reply:
Ah yeah good point! I completely forgot. I'll put this back into WIP status until I've added them.

Reply:
Actually, I might hide them until next time as they're a bit of work, let's see..

Reply:
Awesome 👍